gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/ExploreService.java

```diff
@@ -91,7 +96,8 @@ public ExploreResponse explore(RequestContext requestContext, ExploreRequest req
               newExploreRequestContext, request.getContext());
       exploreRequestValidator.validate(request, attributeMetadataMap);
 
-      IRequestHandler requestHandler = getRequestHandler(request, attributeMetadataMap);
+      IRequestHandler requestHandler =
+          getRequestHandler(request, attributeMetadataMap, requestContext.getTenantId());
```

Comment:
generally we should only send request context and can use contextual key as the key for cache.